File: platforms/ios/Dezel/Dezel/Modules/Graphic/ImageLiveCache.swift
/**
 * Image in memory cache.
 * @class ImageLiveCache
 * @since 0.1.0
 */
open class ImageLiveCache {

	//--------------------------------------------------------------------------
	// MARK: Properties
	//--------------------------------------------------------------------------

	/**
	 * @property cache
	 * @since 0.1.0
	 * @hidden
	 */
	private(set) public var cache: NSCache = NSCache<NSString, UIImage>()

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * @method set
	 * @since 0.1.0
	 * @hidden
	 */
	open func set(_ uri: String, data: UIImage) {
		self.cache.setObject(data, forKey: uri as NSString)
	}

	/**
	 * @method get
	 * @since 0.1.0
	 * @hidden
	 */
	open func get(_ uri: String) -> UIImage? {
		return self.cache.object(forKey: uri as NSString)
	}

	/**
	 * @method has
	 * @since 0.1.0
	 * @hidden
	 */
	open func has(_ uri: String) -> Bool {
		return self.get(uri) != nil
	}
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/modules/graphic/ImageLiveCache.kt
package ca.logaritm.dezel.modules.graphic

import android.graphics.Bitmap
import android.util.LruCache

/**
 * @class ImageLiveCache
 * @since 0.1.0
 * @hidden
 */
open class ImageLiveCache(val size: Int) {

	//--------------------------------------------------------------------------
	// Properties
	//--------------------------------------------------------------------------

	/**
	 * @property cache
	 * @since 0.1.0
	 * @hidden
	 */
	private val cache: LruCache<String, Bitmap> = LruCache(size)

	//--------------------------------------------------------------------------
	// Methods
	//--------------------------------------------------------------------------

	/**
	 * Stores the bitmap in memory.
	 * @method set
	 * @since 0.1.0
	 */
	public fun set(uri: String, bitmap: Bitmap) {
		if (this.cache.get(uri) == null) {
			this.cache.put(uri, bitmap)
		}
	}

	/**
	 * Retrieves the bitmap from memory.
	 * @method get
	 * @since 0.1.0
	 */
	public fun get(uri: String): Bitmap? {
		return this.cache.get(uri)
	}

	/**
	 * Indicates whether a memory cache exists.
	 * @method has
	 * @since 0.1.0
	 */
	public fun has(uri: String): Boolean {
		return this.cache.get(uri) != null
	}
}